### PR TITLE
Fixed alias in Cloudwatch Expressions

### LINKED ISSUE #15562: CloudWatch DataSource Alias is ignored
### What Grafana version are you using?
5.4.3
### What datasource are you using?
CloudWatch
### What OS are you running grafana on?
Linux
### What did you do?
I added a CloudWatch query with an expression
### What was the expected result?
The alias should be used
### What happened instead?
The alias is ignored

The CloudWatch datasource ignores the alias. If an alias is defined, the `formatAlias` method is skipped.

https://github.com/grafana/grafana/blob/653918056c594d7f56a65771b2c9681bdf8a3b9a/pkg/tsdb/cloudwatch/cloudwatch.go#L520-L523

Is this behavior intended? If yes, the alias input field should be removed. If no, I would propose removing this check and using the `queryId` only as a fallback.

I can create a PR but wanted to check whether this is a bug or not.
 
Thanks!
Tarek

### PR DESCRIPTION
**Pull request by utkarshcmu**
closes #15562

### CODE REVIEW

**Comment by tinder-ubhatnagar** [comment]
@mtanda - Review please!

File: pkg/tsdb/cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -496,9 +496,6 @@ func parseQuery(model *simplejson.Json) (*CloudWatchQuery, error) {
 	}
 
 	alias := model.Get("alias").MustString()
-	if alias == "" {
-		alias = "{{metric}}_{{stat}}"
```
Comment (mtanda) [inline pkg/tsdb/cloudwatch/cloudwatch.go]: Need to set default alias in somewhere.

Please check https://github.com/grafana/grafana/pull/9752 .

File: pkg/tsdb/cloudwatch/cloudwatch.go [hunk 2]
```diff
@@ -521,7 +518,11 @@ func parseQuery(model *simplejson.Json) (*CloudWatchQuery, error) {
 
 func formatAlias(query *CloudWatchQuery, stat string, dimensions map[string]string) string {
 	if len(query.Id) > 0 && len(query.Expression) > 0 {
-		return query.Id
+		if len(query.Alias) > 0 {
```
Comment (mtanda) [inline pkg/tsdb/cloudwatch/cloudwatch.go]: I suggest to change default alias of the case `len(query.Id) > 0` and `len(query.Expression) == 0` and `len(query.Alias) == 0`.

This is a suggested change. Please check :-)
https://github.com/grafana/grafana/commit/c5556954739b7a2e10f4eeaa548f9d6f5f67943f

**Review by mtanda — CHANGES_REQUESTED** [review]

File: pkg/tsdb/cloudwatch/cloudwatch.go (hunk 1 above)
Comment (tinder-ubhatnagar) [reply pkg/tsdb/cloudwatch/cloudwatch.go]: I don't think we need a default value as when there is no alias defined, we are displaying the query as it is and handling the case on line 524. By assigning a default value, we are adding a complexity just for the sake of assigning a default value to a variable. By default in go, string values are empty string so we handle that in the else condition on line 524. @bergquist - Any suggestions?

File: pkg/tsdb/cloudwatch/cloudwatch.go (hunk 2 above)
Comment (tinder-ubhatnagar) [reply pkg/tsdb/cloudwatch/cloudwatch.go]: Same comment as above - I don't think we need a default value as when there is no alias defined, we are displaying the query as it is and handling the case on line 524. By assigning a default value, we are adding a complexity just for the sake of assigning a default value to a variable. By default in go, string values are empty string so we handle that in the else condition on line 524. @bergquist - Any suggestions?

File: pkg/tsdb/cloudwatch/cloudwatch.go (hunk 1 above)
Comment (tinder-ubhatnagar) [reply pkg/tsdb/cloudwatch/cloudwatch.go]: @mtanda Any comments?

File: pkg/tsdb/cloudwatch/cloudwatch.go (hunk 2 above)
Comment (tinder-ubhatnagar) [reply pkg/tsdb/cloudwatch/cloudwatch.go]: @mtanda Any comments?

File: pkg/tsdb/cloudwatch/cloudwatch.go (hunk 1 above)
Comment (mtanda) [reply pkg/tsdb/cloudwatch/cloudwatch.go]: The current behavior is showing some legend defined by default alias setting. Incompatibility will cause confusion, it should announce in the release note.

I can't judge which behavior is better and don't have a strong opinion.

**Review by mtanda — APPROVED** [review]

**Comment by tinder-ubhatnagar** [comment]
Thanks @mtanda for approving this PR. Can we get this merged please? Thanks!

**Comment by torkelo** [comment]
@utkarshcmu we have reports that labels are missing, https://github.com/grafana/grafana/issues/16605

**Pull request merged**